### ISSUE TITLE
detect ignored exitcode from subshell parameter substitution

### ISSUE THREAD

**Issue by emdej** [koalaman/shellcheck#2484]
#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -Eeuo pipefail

printf "%s" "$(false)"
echo "went fine"

cat <<EOF
$(false)
EOF
echo "went fine"


```

#### Here's what shellcheck currently says:
```sh
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:
```sh
Exit code from subshell ignored.
```

#### Recommended alternative can be to use a temporary variable for that, like:
```sh
X="$(command)"
printf "%s" "$X"
```
or even better
```sh
local X
X="$(command)"
printf "%s" "$X"
```
instead of
```sh
printf "%s" "$(command)"
```

**Comment by DoxasticFox** [comment]
This issue was resolved by https://github.com/koalaman/shellcheck/pull/2320.

**Comment by brother** [comment]
Enabling the [optional](https://github.com/koalaman/shellcheck/wiki/Optional) `check-extra-masked-returns` will make the example code in TS trigger the warnings.
